decidim-core/app/packs/src/decidim/input_emoji.js

```diff
@@ -1,64 +1,153 @@
-import { screens } from "tailwindcss/defaultTheme"
-import { SUPPORTED_LOCALES } from "emojibase";
+import data from "@emoji-mart/data"
+import i18nEn from "@emoji-mart/data/i18n/en.json"
+import { Picker } from "emoji-mart"
 
 import * as i18n from "src/decidim/i18n";
 
-let I18N_CONFIG = null;
-
-export class EmojiButton {
-  static i18n() {
-    if (I18N_CONFIG) {
-      return I18N_CONFIG;
+class EmojiI18n {
+  static isObject(item) {
+    return (item && typeof item === "object" && !Array.isArray(item));
+  }
+  static deepMerge(target, ...sources) {
+    if (!sources.length) {
+      return target;
     }
-
-    let dict = i18n.getMessages("emojis") || null;
-    const buttonText = dict.button;
-    if (dict) {
-      Reflect.deleteProperty(dict, "button");
-      dict = i18n.createDictionary(dict);
+    const source = sources.shift();
+
+    if (this.isObject(target) && this.isObject(source)) {
+      for (const key in source) {
+        if (this.isObject(source[key])) {
+          if (!target[key]) {
+            Object.assign(target, { [key]: {} });
+          }
+          this.deepMerge(target[key], source[key]);
+        } else {
+          Object.assign(target, { [key]: source[key] });
+        }
+      }
     }
 
-    // dictionary = the messages dictionary passed to Picmo
-    // messages = local "extra" messages
-    I18N_CONFIG = {
-      dictionary: dict,
-      messages: { buttonText }
-    }
-    return I18N_CONFIG;
+    return this.deepMerge(target, ...sources);
   }
 
-  // Get the current locale used for the emoji database
-  //
-  // @returns {string} the current locale if it is supported by emoji base, or english as the fallback locale
   static locale() {
-    let emojiLocale = document.documentElement.getAttribute("lang");
+    return document.documentElement.getAttribute("lang");
+  }
 
-    if (!SUPPORTED_LOCALES.includes(emojiLocale)) {
-      const secondaryLocale = emojiLocale?.split("-")[0];
-      if (SUPPORTED_LOCALES.includes(secondaryLocale)) {
-        emojiLocale = secondaryLocale;
-      } else {
-        emojiLocale = "en";
-      }
+  static i18n() {
+    return this.deepMerge(i18nEn, i18n.getMessages("emojis"));
+  }
+}
+class EmojiPopUp {
+
+  constructor(pickerOptions) {
+    this.popUp = this.createContainer();
+    this.popUp.appendChild(this.createCloseButton());
+    this.popUp.appendChild(this.addStyles());
+
+    let container = document.createElement("div");
+
+    this.picker = new Picker({
+      parent: container,
+      i18n: EmojiI18n.i18n(),
+      locale: EmojiI18n.locale(),
+      data: data,
+      perLine: 8,
+      emojiButtonSize: 41,
+      emojiSize: 30,
+      theme: "light",
+      ...pickerOptions
+    });
+
+    this.popUp.appendChild(container);
+
+    this.setCoordinates();
+  }
+
+  createCloseButton() {
+    let closeButton = document.createElement("button");
+    closeButton.type = "button";
+    closeButton.classList.add("picmo__closeButton");
```

Comment:
Can we change this to something generic? I'd expect to not see any reference to picmo anymore
```suggestion
    closeButton.classList.add("emoji-picker__closeButton");
```